CIP-????/README.md

```diff
@@ -0,0 +1,294 @@
+| Ed25519                 |                 64 |                64 |                                   1 |    [Draft](https://www.nist.gov/news-events/news/2019/10/digital-signatures-and-elliptic-curve-cryptography-request-comments-draft) |
+| Falcon512               |                897 |               666 |                                 0.5 | [Selected](https://csrc.nist.gov/projects/post-quantum-cryptography/selected-algorithms-2022) | 
+| Dilithium2              |               1312 |              2420 |                                 0.5 | [Selected](https://csrc.nist.gov/projects/post-quantum-cryptography/selected-algorithms-2022) |
+| Sphincs+                |                 32 |              8080^[https://sofiaceli.com/2022/07/05/pq-signatures.html#:~:text=A%20table%20of-,comparison%20of%20sizes,-%3A] |                                 2.8^[Benchmark was for Sphins\*128s] | [Selected](https://csrc.nist.gov/projects/post-quantum-cryptography/selected-algorithms-2022) |
+| sequoa863pq128          |                108 |               216 |                                    ? | Submitted^[Communication on ebats@cryp.to mailing list.] |
+
+  : Reference for signature, key sizes, and time cost multipliers [as suggested here](https://blog.cloudflare.com/nist-post-quantum-surprise/). In the order of preference, with comparison against current default signature algorithm (Ed25519).
+
+The following function should then be used to verify arbitrary binary signed
+with permitted signature scheme:
+
+```
+verifySignWithAlgo :: SignAlgo
+                  -> BuiltinByteString -- ^ public key
+                  -> BuiltinByteString -- ^ message or hash
+                  -> BuiltinByteString -- ^ signature
+                  -> BuiltinBool
+```
+
+This function takes parameters of a specific part of the signature scheme, even
+though they are all encoded as `BuiltinByteString`s. In order, for both
+functions, these are:
+
+1. A verification key (length depends on the algorithm chosen);
+2. An input to verify (either the message itself, and a hash);
+3. A signature.
+
+The two different schemes handle deserialization internally: specifically, there
+is a distinction made between 'external' representations, which are expected as
+arguments, and 'internal' representations, used only by the implementations
+themselves. This creates different expecations for each argument for both of
+these schemes; we describe these below.
+
+For the [all signature
+schemes](),
+the keys are required to be given in the same format as NIST test vectors.
+
+The builtin operations will error with a descriptive message if given inputs
+that don't correspond to the constraints above, return `False` if the signature
+fails to verify the input given the key, and `True` otherwise.
+
+### Post-quantum wallets
+
+We should permit using all chosen signature algorithms to lock transactions.
```

Comment:
:rotating_light: :rotating_light: :rotating_light: 

I assume you mean "to lock UTXOs".

This is A HUGE CHANGE and not a Plutus one. This is now veering into changing all the cryptography in Cardano as a whole. I _highly_ recommend making a separate CIP for that, as I think it will be extremely difficult and contentious and needs a lot of support.